src/sync.rs

```diff
@@ -677,6 +693,40 @@ impl<T: Copy> LockFreeFrozenVec<T> {
         let local_index = index - prior_total_buffer_size(buffer_idx);
         unsafe { *buffer_ptr.add(local_index) }
     }
+
+    /// Run a function on each buffer in the vector.
+    ///
+    /// ## Arguments
+    /// - `func`: a function that takes a pointer to the buffer, the buffer size, and the buffer index
+    ///
+    /// ## Safety
+    /// `func` is provided with the raw constant pointer to the buffer,
+    /// however, the pointer is expected to be valid as the null check is done before calling `func`.
+    /// To access the data in the buffer, make sure the buffer size (the second argument) is respected.
+    ///
+    pub unsafe fn for_each_buffer(&self, mut func: impl FnMut(*const T, usize, usize)) {
```

Comment:
You could use `NoNull<T>` instead of `*const T`, or even return `&[MaybeUninit<T>]`. Creating the slice is zero cost, but signals very useful information.

Reply:
Fixed in [`b34ae88` (#57)](https://github.com/Manishearth/elsa/pull/57/commits/b34ae88cfce765c0baa629758563fdd3434a767c)